aws/resource_aws_mwaa_environment.go

```diff
@@ -0,0 +1,753 @@
+	}
+
+	log.Printf("[INFO] Creating MWAA Environment: %s", input)
+	out, err := conn.CreateEnvironment(&input)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(aws.StringValue(input.Name))
+	d.Set("arn", out.Arn)
+
+	if err := waitForMwaaEnvironmentCreation(conn, d.Id()); err != nil {
+		return err
+	}
+
+	return resourceAwsMwaaEnvironmentRead(d, meta)
+}
+
+func resourceAwsMwaaEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mwaaconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	log.Printf("[INFO] Reading MWAA Environment: %s", d.Id())
+
+	out, err := conn.GetEnvironment(&mwaa.GetEnvironmentInput{
+		Name: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if isAWSErr(err, mwaa.ErrCodeResourceNotFoundException, "") && !d.IsNewResource() {
+			log.Printf("[WARN] MWAA Environment %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+
+		return err
+	}
+
+	d.Set("airflow_configuration_options", aws.StringValueMap(out.Environment.AirflowConfigurationOptions))
+	d.Set("airflow_version", out.Environment.AirflowVersion)
+	d.Set("arn", out.Environment.Arn)
+	d.Set("created_at", aws.TimeValue(out.Environment.CreatedAt).String())
+	d.Set("dag_s3_path", out.Environment.DagS3Path)
+	d.Set("environment_class", out.Environment.EnvironmentClass)
+	d.Set("execution_role_arn", out.Environment.ExecutionRoleArn)
+	d.Set("kms_key", out.Environment.KmsKey)
+	if err := d.Set("last_updated", flattenMwaaLastUpdate(out.Environment.LastUpdate)); err != nil {
+		return err
+	}
+	if err := d.Set("logging_configuration", flattenMwaaLoggingConfiguration(out.Environment.LoggingConfiguration)); err != nil {
+		return err
+	}
+	d.Set("max_workers", out.Environment.MaxWorkers)
+	d.Set("name", out.Environment.Name)
+	if err := d.Set("network_configuration", flattenMwaaNetworkConfiguration(out.Environment.NetworkConfiguration)); err != nil {
+		return err
+	}
+	d.Set("plugins_s3_object_version", out.Environment.PluginsS3ObjectVersion)
+	d.Set("plugins_s3_path", out.Environment.PluginsS3Path)
+	d.Set("requirements_s3_object_version", out.Environment.RequirementsS3ObjectVersion)
+	d.Set("requirements_s3_path", out.Environment.RequirementsS3Path)
+	d.Set("service_role_arn", out.Environment.ServiceRoleArn)
+	d.Set("source_bucket_arn", out.Environment.SourceBucketArn)
+	d.Set("status", out.Environment.Status)
+	d.Set("webserver_access_mode", out.Environment.WebserverAccessMode)
+	d.Set("webserver_url", out.Environment.WebserverUrl)
+	d.Set("weekly_maintenance_window_start", out.Environment.WeeklyMaintenanceWindowStart)
+
+	if err := d.Set("tags", keyvaluetags.MwaaKeyValueTags(out.Environment.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsMwaaEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).mwaaconn
+
+	needsUpdate := false
```

Comment:
Should be able to use `if d.HasChangesExcept("tags") {}` block instead of using a boolean flag: #16627.